src/components/AR/AR-session.ts

```diff
@@ -0,0 +1,97 @@
+import { ARProvider } from "./AR-provider";
+
+/**
+ * ARSession - master control for the AR session.
+ * - loads dependencies (aka providers)
+ * - handles global callbacks when AR session is started, ended
+ * - can end any running AR session.
+ * - renders AR button when the scene is loaded 
+ * 
+ * TODO - refactor checkSceneLoadProgress whenever we can control when the WL.onSceneLoaded is fired.
+ * Currently we listen to WL.onSceneLoaded AND to each registered provide's loading to decide when to show the AR button.
+ * Instead, we want to show the engines native 'loading' screen until all the providers are loaded  
+ */
+abstract class ARSession {
+  // tracking provider is basically a lib which has some tracking capabilities, so device native webXR, 8thwall, mind-ar-js, etc
+  private static trackingProviders: Array<ARProvider> = [];
+
+  // current running provider
+  private static currentTrackingProvider: ARProvider | null = null;
+
+  public static readonly onARSessionReady: Array<() => void> = [];
```

Comment:
For every event, you will need to use the new `Emitter` class from `wonderlandengine/api`. It's not published **yet** but will soon be. Basically you should use that for events anywhere you have events. I am leaving this comment so we don't forget to do it